lms/djangoapps/courseware/access.py

```diff
@@ -345,11 +347,7 @@ def can_load_forum():
         """
         Can this user access the forums in this course?
         """
-        return (
-            can_load()
-            and
-            UserProfile.has_registered(user)
-        )
+        return can_load() and user.is_authenticated()
```

Comment:
As a potential follow-up, apply the `require_non_lazy` decorator to the forum views.